src/arch/MovieTexture/MovieTexture_FFMpeg.cpp

```diff
@@ -164,93 +166,128 @@ float MovieDecoder_FFMpeg::GetTimestamp() const
 	}
 
 	// In a logical situation, this means that display is outpacing decoding.
-	if (display_frame_num_ >= static_cast<int>(frame_buffer_.size())) {
+	if (display_frame_num_ >= static_cast<int>(packet_buffer_.size())) {
+		return 0;
+	}
+
+	PacketHolder* packet = packet_buffer_[display_frame_num_].get();
```

Comment:
```
std::vector<std::unique_ptr<PacketHolder>>::iterator it = std::next(packet_buffer_.begin(), display_frame_num_);

PacketHolder* packet = it->get();
```

would work for a `std::list`

`auto` would work here of course but it's preferred to use explicit data types in this code base